File: cakk-domain/src/main/java/com/cakk/domain/repository/jpa/CakeCategoryJpaRepository.java
package com.cakk.domain.repository.jpa;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.cakk.domain.entity.cake.CakeCategory;

public interface CakeCategoryJpaRepository extends JpaRepository<CakeCategory, Long> {

	Optional<CakeCategory> findByCakeId(Long cakeId);
}

File: cakk-infrastructure/persistence-mysql/src/main/java/com/cakk/infrastructure/persistence/repository/jpa/CakeCategoryJpaRepository.kt
package com.cakk.infrastructure.persistence.repository.jpa

import org.springframework.data.jpa.repository.JpaRepository
import com.cakk.infrastructure.persistence.entity.cake.CakeCategoryEntity

interface CakeCategoryJpaRepository : JpaRepository<CakeCategoryEntity, Long> {

	fun findByCakeId(cakeId: Long?): CakeCategoryEntity?
}

